pkg/distributor/errors_test.go

```diff
@@ -120,137 +120,201 @@ func TestNewRequestRateError(t *testing.T) {
 	checkDistributorError(t, wrappedErr, mimirpb.REQUEST_RATE_LIMITED)
 }
 
+func TestNewIngesterPushError(t *testing.T) {
+	testMsg := "this is an error"
+	anotherTestMsg := "this is another error"
+	tests := map[string]struct {
+		originalStatus *status.Status
+		expectedCause  mimirpb.ErrorCause
+	}{
+		"a gRPC error with details give an ingesterPushError with the same details": {
+			originalStatus: createGRPCErrorWithDetails(t, codes.Internal, testMsg, mimirpb.SERVICE_UNAVAILABLE),
```

Comment:
This is a bit confusing to me since this is an `error` but we're treating it as `*status.Status`?

Reply:
`createGRPCErrorWithDetails()` has been renamed into `createStatusWithDetails()`.